a3fe/tests/test_run_integration.py

```diff
@@ -0,0 +1,214 @@
+            input_dir=f"{temp_dir}/input",
+            ensemble_size=2,
+            stream_log_level=logging.CRITICAL,
+        )  # Close the log output
+        calc._dump()
+        # Must use yield so that the temporary directory is deleted after the tests
+        # by the context manager and does not persist
+        yield calc
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(not SLURM_PRESENT, reason="SLURM not present")
+@pytest.mark.skipif(not RUN_SLURM_TESTS, reason="RUN_SLURM_TESTS is False")
+class TestSlurmIntegration:
+    """Test class for SLURM integration tests."""
+
+    def _get_test_config(self, system_prep_config):
+        """Get the system preparation configuration for testing"""
+        cfg = system_prep_config()
+        cfg.runtime_npt_unrestrained = 20  # ps very short for testing
+        cfg.runtime_npt = 20  # ps
+        cfg.ensemble_equilibration_time = 20  # ps
+        return cfg
+
+    def _setup_calculation(self, calc, system_prep_config):
+        """Check that the calculation object is setup already."""
+        if not calc.setup_complete:
+            cfg = self._get_test_config(system_prep_config)
+            calc.setup(bound_leg_sysprep_config=cfg, free_leg_sysprep_config=cfg)
+
+        assert calc.setup_complete
+        return calc
+
+    @pytest.mark.integration
+    def test_slurm_calculation_setup(self, slurm_calc, system_prep_config):
+        """Test that the SLURM calculation setup works correctly."""
+
+        calc = self._setup_calculation(slurm_calc, system_prep_config)
+
+        # Check that the required SLURM job names exist
+        required_slurm_jobnames = ["minimised.sh", "solvated.sh", "preequilibrated.sh"]
+        for jobname in required_slurm_jobnames:
+            job_path = os.path.join(calc.base_dir, "input", jobname)
+            assert os.path.exists(job_path)
+
+        # Check that the required leg directories exist
+        for leg in LEGS_WITH_STAGES.keys():
+            leg_dir = os.path.join(calc.base_dir, leg)
+            assert os.path.exists(leg_dir)
+
+            # Check that the required stage directories exist
+            for stage in LEGS_WITH_STAGES[leg]:
+                stage_dir = os.path.join(leg_dir, stage)
+                assert os.path.exists(stage_dir)
+                output_dir = os.path.join(stage_dir, "output")
+                assert os.path.exists(output_dir)
+
+    @pytest.mark.integration
+    def test_slurm_non_adaptive_run(self, slurm_calc, system_prep_config):
+        """
+        Full integration test for non-adaptive run using SLURM.
+        """
+        calc = self._setup_calculation(slurm_calc, system_prep_config)
+
+        # Run a short non-adaptive calculation
+        calc.run(adaptive=False, runtime=0.1)
+        calc.wait()
+
+        # Check that the calculation is not running
+        assert not calc.running
+
+        # Set the entire simulation time to equilibrated for analysis
+        for leg in calc.legs:
```

Comment:
I used to have to do this before `calc.set_equil` or whatever it is was introduced. Please use this function instead.